### ISSUE TITLE
Build failure on appveyor that is new in 1.21.6

### ISSUE THREAD

**Issue by mwcraig** [conda/conda-build#1130]
I tried out 1.21.6 tonight and got a build fail that I do not get with 1.21.5 (though the version of 1.21.5 I'm using is patched to take pin-numpy and setup-options).

A build that fails is at: https://ci.appveyor.com/project/Astropy/conda-channel-astropy/build/1.0.172#L1183

If you look at the history of the builds for that project, the one immediately before it, which uses 1.21.5, passes (I tried switching back and forth a couple times in case it was a fluke).

**Comment by msarahan** [comment]
Yikes, something is trashing os.environ.  Looks like something that is fine for a single build, but bad for conda-build-all.  I have encountered things like this while playing with #1119, but had not seen it with 1.21.6.  I'll look into it.

**Comment by ocefpaf** [comment]
I believe https://github.com/conda-forge/staged-recipes/pull/1111#issuecomment-234246070 is probably related to this, right?

**Comment by msarahan** [comment]
Yeah, looks related.  Could be an issue in either conda-build or conda-build-all - but given that conda-build 1.21.5 works, perhaps more the former.

**Comment by ocefpaf** [comment]
If you need a test case let me know and I can try it on https://github.com/conda-forge/humanfriendly-feedstock/pull/1

**Comment by patricksnape** [comment]
Must be something to do with [these changes](https://github.com/conda/conda-build/compare/1.21.5...1.21.6#diff-ac55d744935b1f37a3369258b88e8f1fR735) since they are the only thing that messes with the environ?

**Comment by msarahan** [comment]
yes, @patricksnape that seems likely.  At the least, I should limit my env modifications to PATH.

**Comment by jakirkham** [comment]
Yep, also seeing this.

**Comment by jakirkham** [comment]
Should add I see this even with builds that don't use `numpy` or `python`. For instance, this [one]( https://ci.appveyor.com/project/conda-forge/staged-recipes/build/1.0.4872/job/9rt89ool25vjcv6n ) just doesn't have dependencies. Same error message though.

```
source tree in: C:\Miniconda\conda-bld\work\nsis-2.46
Traceback (most recent call last):
  File "C:\Miniconda\Scripts\conda-build-all-script.py", line 9, in <module>
    load_entry_point('conda-build-all==0.13.0', 'console_scripts', 'conda-build-all')()
  File "C:\Miniconda\lib\site-packages\conda_build_all\cli.py", line 85, in main
    b.main()
  File "C:\Miniconda\lib\site-packages\conda_build_all\builder.py", line 224, in main
    built_dist_location = self.build(meta)
  File "C:\Miniconda\lib\site-packages\conda_build_all\builder.py", line 181, in build
    return bldpkg_path(build.build(meta.meta))
  File "C:\Miniconda\lib\site-packages\conda_build_all\build.py", line 32, in build
    build_module.build(meta, post=None, need_source_download=True, **kwd)
  File "C:\Miniconda\lib\site-packages\conda_build\build.py", line 573, in build
    windows.build(m, build_file, dirty=dirty, activate=activate)
  File "C:\Miniconda\lib\site-packages\conda_build\windows.py", line 207, in build
    fo.write(msvc_env_cmd(bits=cc.bits, override=m.get_value('build/msvc_compiler', None)))
  File "C:\Miniconda\lib\site-packages\conda_build\windows.py", line 114, in msvc_env_cmd
    vcvarsall_vs_path = build_vcvarsall_vs_path(version)
  File "C:\Miniconda\lib\site-packages\conda_build\windows.py", line 75, in build_vcvarsall_vs_path
    PROGRAM_FILES_PATH = os.environ['ProgramFiles']
KeyError: 'ProgramFiles'
Command exited with code 1
```

**Comment by msarahan** [comment]
Fixed in #1135 - closing for now.
